react-client/src/Components/DomInfoDialogBox/index.js

```diff
@@ -121,18 +128,47 @@ const DomInfoDialogBox = ({ id, idx, tag, classNames, classNamesString, parent,
   React.useEffect(() => {
     // if the DOM has existing bookmark, enable it by default.
     setStateHasExistingBookmark(hasExistingBookmark);
-
     return () => {
-      
     }
   }, [hasExistingBookmark]);
 
+  React.useEffect(() => {
+    setStateHasExistingAnnotation(hasExistingAnnotations);
+    return () => {      
+    }
+  }, [hasExistingAnnotations]);  
+  
+  useEffect(() => {
+    hasExistingAnnotations = domUtils.hasAnnotations(annotationStore, domElement);
+    setStateHasExistingAnnotation(hasExistingAnnotations);                  
+    return () => {      
+    }
+  }, [annotationStore])
+  
   const onClickAddBookmark = () => { 
-
     setStateHasExistingBookmark(!stateHasExistingBookmark);
     setShowAddBookmarkPanel(!showAddBookmarkPanel);
+    setShowAddAnnotationsPanel(false);
+  };
+
+  const onClickAddAnnotation = () => { 
+    if (stateHasExistingAnnotation === false) {
```

Comment:
I feel like this is redundant. Why bother doing this:

`if (stateHasExistingAnnotation === false) {
      setStateHasExistingAnnotation(!stateHasExistingAnnotation);
 }`

Reply:
yes this is needed since if we are adding new annotation using just the  useEffect with that is monitoring changes in annotationStore would result in false since there is no data to be found this would force the icon to be shaded if we are enetering a new annotation for the element

Reply:
had modified  the codes to fit better but yes this function is necessary.  we cannot just depend on querying from local storage to change the state of a disabled annotation icon since if it is a new record then querying from local storage would always return as false

Reply:
Okay in that case you can use this logic instead to make it cleaner:

`if(!stateHasExistingAnnotation)`

So you can read the code like: **if not stateHasExistingAnnotation**

Reply:
yep i used     !stateHasExistingAnnotation ? setStateHasExistingAnnotation(!stateHasExistingAnnotation) : getAnnotationStoreUpdates();    
got the idea from corrections in addannotationpanel.js

Reply:
Looks good